src/App.tsx

```diff
@@ -1,24 +1,248 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { createTodos, getTodos, removeTodos } from './api/todos';
+import { TodoFilter } from './components/TodoFilter';
+import { ErrorType } from './types/ErrorMessage';
 
-const USER_ID = 0;
+const USER_ID = 11356;
+
+function getFilteredTodos(
+  todos: Todo[],
+  sortBy: string,
+) {
+  const filteredTodos = todos;
+
+  switch (sortBy) {
+    case Status.ALL:
+      return filteredTodos;
+
+    case Status.ACTIVE:
+      return filteredTodos.filter(todo => todo.completed === false);
+
+    case Status.COMPLETED:
+      return filteredTodos.filter(todo => todo.completed === true);
+
+    default:
+      return filteredTodos;
+  }
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<string>(Status.ALL);
+  const [errorMessage, setErrorMessage] = useState(ErrorType.None);
+  const [isMessageClosed, setIsMessageClosed] = useState(false);
+  const [newTitle, setNewTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isDeleting, setIsDeleting] = useState<number[]>([]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorType.LoadingError);
+        setIsMessageClosed(false);
+        setTimeout(() => {
+          setIsMessageClosed(true);
+        }, 3000);
+      });
+  }, [todos]);
```

Comment:
here you problem, you get todos -> set it -> useEffect fired again (because of it dependency)